src/test/java/org/sagebionetworks/bridge/udd/helper/ZipHelperTest.java

```diff
@@ -90,4 +136,25 @@ public void testZip() throws Exception {
             return unzippedMap;
         }
     }
+
+    public static Map<String, String> unzipHelper(byte[] zipBytes, char[] password) {
+        Map<String, String> unzippedMap = new HashMap<>();
+
+        try (ByteArrayInputStream zipBytesInputStream = new ByteArrayInputStream(zipBytes);
+             net.lingala.zip4j.io.inputstream.ZipInputStream zipInputStream =
+                     new net.lingala.zip4j.io.inputstream.ZipInputStream(zipBytesInputStream, password)) {
+            LocalFileHeader localFileHeader = zipInputStream.getNextEntry();
+            while (localFileHeader != null) {
+                String fileName = localFileHeader.getFileName();
+                byte[] fileBytes = ByteStreams.toByteArray(zipInputStream);
+                String fileContent = new String(fileBytes, Charsets.UTF_8);
+                unzippedMap.put(fileName, fileContent);
+
+                localFileHeader = zipInputStream.getNextEntry();
+            }
+            return unzippedMap;
+        } catch (Exception e) {
+            return unzippedMap;
```

Comment:
If this throws, it probably indicates an error in the test somewhere. We should propagate the error back up and have the test fail.

Reply:
The reason I did this was because LocalFileHeader (zip4j's version of ZipEntry) frustratingly does not return null when there is no more file to read, instead it will throw an error. I spent a few hours grappling with this to no avail and I haven't been able to find a way to deal with this. The other way of unzipping a password protected zip file using this library is done in one line but it extracts it to a directory only. Does this make sense?